### PR TITLE
use depguard to prohibitate importing github.com/golang/protobuf  #876(after rebase)

### PR DESCRIPTION
**Pull request by SpongeBob0318**
**What type of PR is this?**
/kind enhancement

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #876

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/controller/ads/convert_filter_test.go [hunk 1]
```diff
@@ -20,8 +20,9 @@ import (
 	"testing"
 
 	envoy_filters_tcp_proxy "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/tcp_proxy/v3"
-	"github.com/golang/protobuf/ptypes/wrappers"
+	//"github.com/golang/protobuf/ptypes/wrappers"
```
Comment (LiZhenCheng9527) [inline pkg/controller/ads/convert_filter_test.go:23]: remove

File: pkg/controller/workload/workload_hash.go [hunk 2]
```diff
@@ -23,7 +23,8 @@ import (
 	"math"
 	"os"
 
-	"gopkg.in/yaml.v3"
+	//"gopkg.in/yaml.v3"
```
Comment (LiZhenCheng9527) [inline pkg/controller/workload/workload_hash.go:26]: remove

File: pkg/controller/config/config.go [hunk 3]
```diff
@@ -21,7 +21,8 @@ import (
 	"strings"
 
 	config_core_v3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
-	structpb "github.com/golang/protobuf/ptypes/struct"
+	//structpb "github.com/golang/protobuf/ptypes/struct"
```
Comment (LiZhenCheng9527) [inline pkg/controller/config/config.go:24]: remove

Comment (SpongeBob0318) [reply pkg/controller/config/config.go:24]: sure, i will remove it

File: Makefile [hunk 4]
```diff
@@ -117,6 +117,8 @@ install:
 	$(QUIET) make install -C bpf/deserialization_to_bpf_map
 	$(QUIET) make install -C kernel/ko_src
 
+	$(QUIET) go install github.com/OpenPeeDeeP/depguard/cmd/depguard@latest
```
Comment (hzxuzhonghu) [inline Makefile:120]: remove

Comment (SpongeBob0318) [reply Makefile]: ok, i have done it

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

Thank you!

**Pull request merged**